site/js/jquery.console.js

```diff
@@ -37,714 +37,809 @@
+                if (cancelKeyPress) return false;
+            }
+
+
+        });
+
+
+        //after input event get new string, figure out the difference and add the character to the typer.
+        typer.on("input", function(e) {
+            var newValue = typer.val();
+            typer.consoleInsert(difference(typer.oldValue, newValue));
+            typer.val("");
+        })
+
+
+        function isIgnorableKey(e) {
+            // for now just filter alt+tab that we receive on some platforms when
+            // user switches windows (goes away from the browser)
+            return ((e.keyCode == keyCodes.tab || e.keyCode == 192) && e.altKey);
+        };
+
+        ////////////////////////////////////////////////////////////////////////
+        // Rotate through the command history
+        function rotateHistory(n) {
+            if (history.length == 0) return;
+            ringn += n;
+            if (ringn < 0) ringn = history.length;
+            else if (ringn > history.length) ringn = 0;
+            var prevText = promptText;
+            if (ringn == 0) {
+                promptText = restoreText;
+            } else {
+                promptText = history[ringn - 1];
+            }
+            if (config.historyPreserveColumn) {
+                if (promptText.length < column + 1) {
+                    column = promptText.length;
+                } else if (column == 0) {
+                    column = promptText.length;
+                }
+            } else {
+                column = promptText.length;
+            }
+            updatePromptDisplay();
+        };
+
+        function previousHistory() {
+            rotateHistory(-1);
+        };
+
+        function nextHistory() {
+            rotateHistory(1);
+        };
+
+        // Add something to the history ring
+        function addToHistory(line) {
+            history.push(line);
+            restoreText = '';
+        };
+
+        // Delete the character at the current position
+        function deleteCharAtPos() {
+            if (column < promptText.length) {
+                promptText =
+                    promptText.substring(0, column) +
+                    promptText.substring(column + 1);
+                restoreText = promptText;
+                return true;
+            } else return false;
+        };
+
+        function backDelete() {
+            if (moveColumn(-1)) {
+                deleteCharAtPos();
+                updatePromptDisplay();
+            }
+        };
+
+        function forwardDelete() {
+            if (deleteCharAtPos()) {
+                updatePromptDisplay();
+            }
+        };
+
+        function deleteUntilEnd() {
+            while (deleteCharAtPos()) {
+                updatePromptDisplay();
+            }
+        };
+
+        function clearCurrentPrompt() {
+            extern.promptText("");
+        };
+
+        function clearScreen() {
```

Comment:
:information_source: This change is coming from this commit: https://github.com/chrisdone/jquery-console/commit/3d405bf1f1613e5c456159adfb6d1b13512b196a